stacker/commands/stacker/base.py

```diff
@@ -1,10 +1,32 @@
 import argparse
+import threading
+import signal
 from collections import Mapping
+from threading import BoundedSemaphore
 import logging
 
 from ...environment import parse_environment
 
 
+def cancel():
+    """Returns a threading.Event() that will get set when SIGTERM, or
+    SIGINT are triggered. This can be used to cancel execution of threads.
+    """
+    cancel = threading.Event()
+
+    def cancel_execution(signum, frame):
+        cancel.set()
+
+    signal.signal(signal.SIGINT, cancel_execution)
+    signal.signal(signal.SIGTERM, cancel_execution)
+    return cancel
+
+
+def build_semaphore(concurrency):
```

Comment:
Might be cleaner to change this to `build_walker` and return an object for walking the graph (`stacker.dag.walk`/`stacker.dag.walk_threaded`), with the semaphore built-in.

This way, when `--max-parallel=1` we can disable threaded execution entirely, which may be useful for debugging in some cases.